WalletWasabi/BitcoinCore/RpcModels/VerboseTransactionInfo.cs

```diff
@@ -0,0 +1,21 @@
+using System.Collections.Generic;
+using NBitcoin;
+
+namespace WalletWasabi.BitcoinCore.RpcModels
+{
+	public class VerboseTransactionInfo
+	{
+		public VerboseTransactionInfo(uint256 id, IEnumerable<VerboseInputInfo> inputs, IEnumerable<VerboseOutputInfo> outputs)
+		{
+			Id = id;
+			Inputs = inputs;
+			Outputs = outputs;
+		}
+
+		public uint256 Id { get; }
+
+		public IEnumerable<VerboseInputInfo> Inputs { get; }
```

Comment:
I remember having this discussion since 2005 and many developers have very good arguments for implementing properties with different types but, IMO `IEnumerable` is the worst possible alternative. In dotnet BCL you will almost never find classes with IEnumerable properties, most of the times they give you back a `ReadOnlyCollection<T>` or some customs readonly class.

https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/guidelines-for-collections

No few devs return arrays in cases like this one too.

Personally I prefer to use generic ReadOnlyCollection of immutable types because the abstraction is better and the usability/readability is equally good, otherwise take a look at how ugly is the chain of `ElementAt(x).Something.ElementAt(y)` chain is.

Reply:
AFAIK IEnumerable is the standard.

Reply:
There is no standard, there are only different alternatives and a few good recommendations created and used by Microsoft itself in their own products shared in its Design Guidelines:

> ✔️ DO use ReadOnlyCollection<T>, a subclass of ReadOnlyCollection<T>, or **in rare cases** IEnumerable<T> for properties or return values representing read-only collections. 

(enphasis is mine)
 
> In cases where you are sure that the only scenario you will ever want to support is forward-only iteration, you can simply use IEnumerable<T>.

So, `IEnumerable<T>` is, in Microsoft stuff opinion, only for a very special case. Its usage is the exception and not the rule.

Reply:
This things are trends that change IMO, once the recommendation was to use ICollection then IEnumerable then IReadOnly. 
Personally in this case I'd just go with List (as it was before) it does not matter so much (unless there are clear benefits of IEnumerable like aggregating linq queries or delayed enumeration).

Reply:
Okay, I also would prefer List than IEnumerable. The discussion here is not focused on this PR, the idea is this things become conventions that we all have to follow so, it is better to start the discussion.

Reply:
That's the worst choice above all as it enables the collection to be modified.